bin/check-sqs-messages.rb

```diff
@@ -121,9 +128,8 @@ def run
       crit = false
       queues = []
 
-      sqs.queues.with_prefix(config[:prefix]).each do |q|
-        queue_name = q.arn.split(':').last
-        messages = q.approximate_number_of_messages
+      sqs.queues(queue_name_prefix: config[:prefix]).each do |q|
+        messages = sqs.client.get_queue_attributes(queue_url: q.url, attribute_names: ['All']).attributes[config[:metric]].to_i
 
```

Comment:
I get `Check failed to run: undefined local variable or method queue_name` when using a queue prefix.  `queue_name = q.attributes['QueueArn'].split(':').last` would do the trick.